src/openstudio_lib/library/geometry_preview.html

```diff
@@ -2674,6 +2676,83 @@
   }
 }
 
+var adjacencyIssueIds = new Set();
+var updateAdjacencyIssues = function(value) {
+  Array.groupBy = function(arr, prop) {
+      return arr.reduce(function(groups, item) {
+          const val = prop.split('.')
+              .reduce((obj, key) => (obj && obj[key] !== 'undefined') ? obj[key] : undefined, item);
+          groups[val] = groups[val] || [];
+          groups[val].push(item);
+          return groups;
+      }, {});
+  }
+
+  function combinations(a, b) {
+      if (b === undefined) {
+          return a.flatMap((v, i) => a.slice(i).map(w => [v, w]));
+      } else {
+          return a.flatMap((v, i) => b.map(w => [v, w])); 
+      }
+  }
+
+  // As of now Three.js revision 71
+  class Box3 extends THREE.Box3 {
+
+      // Added in revision 84
+      expandByObject(object) {
+          const box = new THREE.Box3();
+          box.setFromObject(object);
+          this.expandByPoint(box.min);
+          this.expandByPoint(box.max);
+      }
+
+      // Added in revision 74
+      intersectsBox(box) {
+          // using 6 splitting planes to rule out intersections.
+          return box.max.x < this.min.x || box.min.x > this.max.x ||
+              box.max.y < this.min.y || box.min.y > this.max.y ||
+              box.max.z < this.min.z || box.min.z > this.max.z ? false : true;
+  
+      }
+  }
+
+  const detectable = scene_objects.filter(o => 
+      ["Wall", "RoofCeiling", "Floor"].includes(o.userData.surfaceTypeMaterialName))
+  const per_space = Array.groupBy(detectable, 'userData.spaceName')
+  const space_boxes = Object.entries(per_space).map(([space, objects]) => {
+      const box = new Box3();
+      box.space = space;
+      box.surfaces = objects;
+      objects.forEach(o => box.expandByObject(o));
+      return box;
+  });
+
+  /*
+    * Implements part of Honeybee's SolveAdj tool licensed under the AGPL 3.0. See:
+    * - https://github.com/ladybug-tools/honeybee-grasshopper-core/blob/27773fe0097188d4be0e160c71a0cda2c54c6f73/honeybee_grasshopper_core/src/HB%20Solve%20Adjacency.py#L190
+    * - https://github.com/ladybug-tools/honeybee-core/blob/ae4e905a78635a7448291ff8dbe94aa2f627a1ae/honeybee/room.py#L925
+    * - https://github.com/ladybug-tools/honeybee-core/blob/ae4e905a78635a7448291ff8dbe94aa2f627a1ae/honeybee/room.py#L867
+    * - https://github.com/ladybug-tools/ladybug-geometry/blob/da34f3397c59bc311004cb6319d534953c8ee9ab/ladybug_geometry/geometry3d/polyface.py#L689
+    */
+  const almost_equal = 1e-3;
```

Comment:
We use 0.001 as the distance threshold, so maybe this would be 1e-6 for the squared distance? 

https://github.com/NREL/OpenStudio/blob/develop/src/model/ThreeJSForwardTranslator.cpp#L165

Reply:
Added on #467